stubs/docutils/docutils/parsers/rst/states.pyi

```diff
@@ -264,130 +290,150 @@ class SpecializedBody(Body):
     grid_table_top = invalid_input
     simple_table_top = invalid_input
     explicit_markup = invalid_input
-    anonymous = invalid_input  # type: ignore[assignment]
+    anonymous = invalid_input
     line = invalid_input
     text = invalid_input
 
 class BulletList(SpecializedBody):
-    blank_finish: Incomplete
-    def bullet(  # type: ignore[override]
-        self, match: Match[str], context: list[str] | None, next_state: str | None
-    ) -> tuple[list[str], str | None, list[str]]: ...
+    def bullet(self, match: Match[str], context: Any, next_state: _NextState) -> tuple[list[Any], _NextState, list[Any]]: ...  # type: ignore[override]
 
 class DefinitionList(SpecializedBody):
-    def text(self, match: Match[str], context: list[str] | None, next_state: str | None) -> tuple[list[str], str, list[str]]: ...  # type: ignore[override]
+    def text(self, match: Match[str], context: Any, next_state: Any) -> tuple[list[str], Literal["Definition"], list[Any]]: ...  # type: ignore[override]
 
 class EnumeratedList(SpecializedBody):
     auto: int
     blank_finish: Incomplete
     lastordinal: Incomplete
-    def enumerator(  # type: ignore[override]
-        self, match: Match[str], context: list[str] | None, next_state: str | None
-    ) -> tuple[list[str], str | None, list[str]]: ...
+
+    def enumerator(self, match: Match[str], context: Any, next_state: _NextState) -> tuple[list[Any], _NextState, list[Any]]: ...  # type: ignore[override]
 
 class FieldList(SpecializedBody):
     blank_finish: Incomplete
+
     def field_marker(  # type: ignore[override]
-        self, match: Match[str], context: list[str] | None, next_state: str | None
-    ) -> tuple[list[str], str | None, list[str]]: ...
+        self, match: Match[str], context: Any, next_state: _NextState
+    ) -> tuple[list[Any], _NextState, list[Any]]: ...
```

Comment:
I would appreciate some input in these cases as-well. The implementing function is:

```python
def field_marker(self, match, context, next_state):
    """Field list field."""
    field, blank_finish = self.field(match)
    self.parent += field
    self.blank_finish = blank_finish
    return [], next_state, []
```

I'm not sure how the `[]` should be annotated ideally. This is another very common pattern which I have always annotated as `list[Any]`. I stayed away from `list[Never]` here, because this will throw an error if the array is appended to (which I can only assume it often is)

Reply:
```
    Transition methods all return a 3-tuple:

    - A context object, as (potentially) modified by the transition method.
    - The next state name (a return value of ``None`` means no state change).
    - The processing result, a list, which is accumulated by the state
      machine.
```